src/packet.ml

```diff
@@ -247,21 +257,48 @@ let encode_protocol proto len =
       buf
   | `Udp -> Cstruct.empty
 
-let encode proto (key, p) =
-  let payload, len =
+let encode proto hmac_len
+    (key, (p : [< `Ack of header | `Control of operation * _ ])) =
+  let hdr = header p in
+  let len =
+    let id_arr_len = id_len * List.length hdr.ack_sequence_numbers in
+    protocol_len proto + 1 + hdr_len hmac_len + id_arr_len
```

Comment:
the 1 is the operation&key byte, isn't it?

Reply:
Yes, correct

Reply:
maybe add a comment behind the number so we don't have to guess in the future? ;)